### DISCUSSION TITLE
Unable to add panel while developing plugin.

### DISCUSSION THREAD

**Discussion by ChopinDavid** [runelite/runelite#16736] [Q&A]
Hello. I am trying to write my first Runelite plugin. I've got most of the functionality down, but am struggling with displaying a UI.

I am trying to display a panel on the right side of the client like so:
![Screenshot 2023-07-01 095334](https://github.com/runelite/runelite/assets/27712906/02e4ae10-a61f-4865-a30f-02928782b0ec)

I've created a panel that looks something like this:
```java
package com.example;

import com.google.inject.Inject;
import net.runelite.client.ui.PluginPanel;

public class TreeRunHelperPanel extends PluginPanel {
	@Inject
	TreeRunHelperPanel() {
		super();
	}
}
```

and my plugin looks something like this:
```java
public class TreeRunHelper extends Plugin
{

	...

	@Override
	protected void startUp() throws Exception
	{

		TreeRunHelperPanel panel = injector.getInstance(TreeRunHelperPanel.class);
		navButton = NavigationButton.builder()
				.tooltip("Tree Run Helper")
				.icon(ImageUtil.loadImageResource(getClass(), "/panel_icon.png"))
				.panel(panel)
				.priority(50)
				.build();

		clientToolbar.addNavigation(navButton);
	}

	...

}
```

When launching the app, I am able to see my `NavigationButton` being added to the `ClientToolbar` via [`ClientToolbar.java lines 55-66`](https://github.com/runelite/runelite/blob/408121c60143bea685532f86228fe6c7c7329624/runelite-client/src/main/java/net/runelite/client/ui/ClientToolbar.java#L55-L66), but the panel is not rendering along the right side of the client.

I am sort of pulling my hair out at this point. Is there a key step that I am missing? Is there maybe a better place to ask this question? Any help is appreciated! **FYI, [here is the repo/branch in question](https://github.com/ChopinDavid/tree-run-helper/tree/TryingToCreateAPanel) if anybody wants a closer look.**

**Comment by raiyni** [comment]
This isn't a help forum, development help could be had in our runelite channel on discord.

**Reply by ChopinDavid** [reply]
Is there at least documentation about displaying panels, or do devs have to cross reference what existing plugins do? [This](https://github.com/runelite/runelite/wiki/Creating-plugin-config-panels) seems to explain how to use the config to get panel components "out of the box", but little around actually implementing a custom panel.
